### DISCUSSION TITLE
iperf - iperf3  with AdvantEDGE

### DISCUSSION THREAD

**Discussion by userH1213** [InterDigitalInc/AdvantEDGE#371] [Q&A] [answered]
Hello,

   I'm using demo1 as a base for my scenario and I have some couple of questions. I was able to redirect the log files at the cloud server app and edge app (ex. -c, iperf -u -s -p $IPERF_CLOUD_SERVER_SERVICE_PORT > /tmp/cloudServer.txt)  
but not with ue app (-c, iperf -u -c $IPERF_CLOUD_SERVER_SERVICE_HOST -p $IPERF_CLOUD_SERVER_SERVICE_PORT -t 600 -l 2M > /tmp/UE2.txt;)

1-How can I redirect the log file at the ue app? 
I tried to use iperf3 since it has more suitable choices, but it didn't work with any of the apps. 
2-Does AdvantEDGE support iperf3 ?
 After I deploy my scenario, each time I go to the Monitor page the database is set on internal, and I have to choose the correct one every time I leave the page and get back to it. I can not add an event and check how it affected the data flow; when I move to the Execute page and return to the the Monitor page, the database is back on internal, no data is showing, the system indicator is red and showing different errors, and I have to change it back to the correct one with that all the previous data is lost. 
3-Is there a way to make the database fixed from the deployment time until termination ?

Thank you

**Comment by dilallkx** [comment]
Hi @userH1213,

Thank you for using our platform and for reaching out!

1) You should be able to redirect iperf output to a file in the container (I modified demo1 and it worked on my side). Do you confirm that your pod runs successfully with the command your provide above and without the redirect?
Note: If your pod fails or crashes you can obtain the logs before the crash using the command: `kubectl logs <pod name> -p`

2) The images we deploy in demo1 use Iperf2. If you want to use Iperf3 you will need to use different container images.

3) You can open 2 different browsers (or tabs) simultaneously with the AdvantEDGE frontend. Alternatively, you can display some of the monitoring dashboards in the Execute page as well by clicking on the `Dashboard` and configuring View 2 with the dashboard you want to observe (e.g. Network Metrics Aggregation). These dashboards will already be configured to use the current sandbox and scenario.

I hope this helps and answers your questions.
Regards!

**Reply by userH1213** [reply]
Thank you dilallkx for the prompt response. 

1- Here is what I noticed: sometimes they work, other times some of them work, most times they don't work. What is strange is that when I check the monitor page they are up and running on the graphs, however on the execute page their status is pending.
![pods-in-sandbox2](https://user-images.githubusercontent.com/79055649/180670859-9fc61848-2c88-471e-84b6-b5e5441f311c.png)

even the running pods are not recognized 

![running-pods-notRecognized](https://user-images.githubusercontent.com/79055649/180671358-d6a3c81e-5c2f-43b3-a26e-50ad837e8b41.png)


![graphs-running](https://user-images.githubusercontent.com/79055649/180661738-04b0faae-3ecc-4b74-8abe-12fde4ad62a7.png)

![execte-page](https://user-images.githubusercontent.com/79055649/180661746-9e6945ac-c0da-4a37-b5b7-24e696bdbf77.png)


2- For my previous question regarding losing the data when changing the monitor page, an easy fix (if someone else was wondering like me) is to change the time from last 1 minute to what suites their needs then they can download the csv files. 
  
Thank you

**Comment by userH1213** [comment]
Here is an update. I run demo1 again, to check and compare the performance of the pods. The first time all pods were running, however the second time some of the pods were pending. Can it be a resource limitation? then I run **kubectl describe node** 
![resources-in-node](https://user-images.githubusercontent.com/79055649/180698868-fb427397-a7d8-4edd-998c-e01f53955b3c.png)

I tried deleting old pods from old sandboxes with kubectl delete pod pod-name, but it gives me error from server not found.

**Reply by dilallkx** [reply]
Hi @userH1213,

Thank you for the update!
It definitely seems like a resource limitation. What are your system specifications?
The recommended HW setup is provided here:
https://interdigitalinc.github.io/AdvantEDGE/docs/setup/env-hw/

Also, when you run `kubectl describe node` do you see any error conditions?

For your question about deleting pods, make sure you use `-n <namespace>` in your command. This is most likely the reason you are getting a 'not found' error.

I hope this helps.
Reagrds!

**Reply by userH1213** [reply]
Thank you dilallkx for all the help.

Adding -n <namespace> to kubectl delete pods solved my issue with deleting unused pods, and saved me the trouble of having to delete all dependencies and cores and deploy them back again every time I reboot my machine.
I'm working on a VM and the resources I've allocated to it is sufficient. You can see it in the result of describe node in the attached picture.
 Laptop HW:  AMD Ryzen 5 4600H(6 cores -12 threads) and the RAM is 32GB


![describeNode](https://user-images.githubusercontent.com/79055649/181995547-474e6f45-1ef2-4420-854e-5e021207b774.png)
![restOfDescribingNode](https://user-images.githubusercontent.com/79055649/181995557-eeb8b40c-cf04-4fb1-9e7e-0c646d91890b.png)




Thank you

**Comment by userH1213** [comment, marked as answer]
Update
I discovered that I only face this problem( some pods are running and others are pending) when I specify min & max CPU for terminal applications. That makes perfect sense because in the documentation it says if you specify min and max cpu the pods will run only when this specified number is available.
